api/src/main/java/com/epam/pipeline/dao/pipeline/StopServerlessRunDao.java

```diff
@@ -38,21 +39,22 @@
     private String updateServerlessRunQuery;
     private String loadAllServerlessRunsQuery;
     private String deleteByRunIdServerlessRunQuery;
+    private String loadServerlessunByRunIdQuery;
 
     @Transactional(propagation = Propagation.MANDATORY)
     public Long createServerlessRunId() {
         return daoHelper.createId(serverlessRunSequenceQuery);
     }
 
-    @Transactional(propagation = Propagation.MANDATORY)
+    @Transactional(propagation = Propagation.REQUIRED)
     public void createServerlessRun(final StopServerlessRun run) {
         final Long id = createServerlessRunId();
         run.setId(id);
         getNamedParameterJdbcTemplate().update(saveServerlessRunQuery,
                 StopServerlessRunParameters.getParameters(run));
     }
 
-    @Transactional(propagation = Propagation.MANDATORY)
+    @Transactional(propagation = Propagation.REQUIRED)
```

Comment:
For most classes we keep `Mandatory` propagation at DAO level, why do we need `Required` in this case?

Reply:
We need `Required` here due to this method called from non-transaction context.
I added a new class `StopServerlessRunManager` to manage transactions.